### PR TITLE
Replace Jaro-Winkler algorithm usage with an internal function

### LINKED ISSUE #1892: Vendor usage of `github.com/xrash/smetrics` for suggestions
The usage of `smetrics.JaroWinkler` should be vendored into this source tree or replaced with an implementation that is within this source tree to support the goal of only requiring the Go standard library.

### PR DESCRIPTION
**Pull request by yigithankarabulut**
## What type of PR is this?

- cleanup

## What this PR does / why we need it:

To support the goal of only requiring the Go standard library. We were using the Jaro-Winkler algorithm for the suggestion from the "xrash/smetrics" package. This function was rewritten to be faithful to stdlib.

## Which issue(s) this PR fixes:

Fixes #1892 

## Testing

go test -run=TestJaroWinkler

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

**Comment by yigithankarabulut** [comment]
Probably the rate limit was reached in the check and failed. The remaining steps were therefore canceled.

**Comment by avorima** [comment]
Please run `go mod tidy` to remove the dependency completely

File: suggestions.go [hunk 1]
```diff
@@ -16,13 +16,89 @@ type SuggestFlagFunc func(flags []Flag, provided string, hideHelp bool) string
 
 type SuggestCommandFunc func(commands []*Command, provided string) string
 
+// jaroDistance is the measure of similarity between two strings. The result is
+// 1 for equal strings, and 0 for completely different strings.
```
Comment (bartekpacia) [inline suggestions.go:20]: I'd add that the result is _between_ 0 or 1. This description at first made me think the output is either 0 or 1, only.

**Review by Juneezee — APPROVED** [review]

**Review by bartekpacia — APPROVED** [review]
LGTM, thank you for contributing :)

**Comment by meatballhat** [comment]
Sorry about the trouble with codecov 😩 I'll get a token set up as soon as I can!

**Review by meatballhat — APPROVED** [review]
Lovely, @yigithankarabulut! Thanks so much 🤘🏼

**Comment by yigithankarabulut** [comment]
@meatballhat You can be sure there will be more to come. Let's go!

**Comment by bartekpacia** [comment]
Go go go! :D

**Pull request merged**